### PR TITLE
Add methods to change a material's composition

### PR DESCRIPTION
**Pull request by Rundas01**
## What
For Addon/Modpack devs, it's unfortunate to not have a way of changing a material's composition. Autogenerated recipes obviously work with the composition and not the chemical formula of a material, so to prevent the need of adding the disable decomposition flag and a new decomposition recipe for materials with changed formula alone, it's more reasonable to just change the composition in the first place.

## Implementation Details
Please check whether I used the ZenMethod annotation correctly.

## Outcome
Some Addon/Modpack devs will be happy.

### CODE REVIEW

File: src/main/java/gregtech/api/unification/material/Material.java [hunk 1]
```diff
@@ -110,6 +130,14 @@ public Material setFormula(String formula, boolean withFormatting) {
         return this;
     }
 
+    @ZenMethod
+    public Material setComponents(MaterialStack... components) {
+        this.materialInfo.setComponents(components);
+        this.chemicalFormula = null;
```
Comment (brachy84) [inline src/main/java/gregtech/api/unification/material/Material.java:136]: There is no point in setting the formula to null here, is there?

File: src/main/java/gregtech/api/unification/material/Material.java [hunk 2]
```diff
@@ -1172,5 +1200,10 @@ private void verifyInfo(MaterialProperties p, boolean averageRGB) {
                 }
             }
         }
+
+        public MaterialInfo setComponents(MaterialStack... components) {
+            this.componentList = ImmutableList.copyOf(Arrays.stream(components).collect(Collectors.toList()));
```
Comment (brachy84) [inline src/main/java/gregtech/api/unification/material/Material.java:1205]: `Arrays.asList(components)` instead of stream. Or use immutable collector

**Review by brachy84 — DISMISSED** [review]

File: src/main/java/gregtech/api/unification/material/Material.java (hunk 1 above)
Comment (Zorbatron) [reply src/main/java/gregtech/api/unification/material/Material.java:136]: If the formula isn't set `null`, then it won't actually calculate the new formula. https://github.com/GregTechCEu/GregTech/blob/586828f5697fbc92ec0613f1f1b6f5dd6a529132/src/main/java/gregtech/api/unification/material/Material.java#L77

Comment (brachy84) [reply src/main/java/gregtech/api/unification/material/Material.java:136]: funny

**Review by bruberu — APPROVED** [review]
This basically seems fine now.

**Pull request merged**